### PR TITLE
Nightly rebase

### PR DESCRIPTION
**Pull request by buffalu**
#### Problem
Rebasing Jito-Solana is annoying and time consuming.

#### Summary of Changes
Automatically rebase Jito-Solana on a periodic basis. Send message on slack during any failures or success.

Fixes #

### CODE REVIEW

File: .github/workflows/rebase.yaml [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+            {
+              "text": "Nightly rebase on branch ${{ matrix.branch }}\nStatus: CI success\nBranch: ${{ env.REBASE_BRANCH}}\nBuild: https://buildkite.com/jito/jito-solana/builds?commit=${{ env.REBASE_SHA }}"
+            }
+        env:
+          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
+      - name: Reset ${{ matrix.branch }} to ${{ env.REBASE_BRANCH }}
+        run: |
+          git checkout ${{ matrix.branch }}
+          git reset --hard ${{ env.REBASE_BRANCH }}
+      - name: Push untouched %{{ matrix.branch }} at a moment in time
```
Comment (buffalu) [inline .github/workflows/rebase.yaml:141]: fix comment

File: .github/workflows/rebase.yaml [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+on:
+  push:
+
+jobs:
+  rebase:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        branch:
+          #          - master
```
Comment (buffalu) [inline .github/workflows/rebase.yaml:12]: uncomment

File: .github/workflows/rebase.yaml [hunk 3]
```diff
@@ -0,0 +1,145 @@
+name: "Rebase Jito-Solana from upstream solana-labs/solana"
+
+on:
+  push:
```
Comment (buffalu) [inline .github/workflows/rebase.yaml:4]: add nightly cron

File: .github/workflows/rebase.yaml [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+        with:
+          ref: ${{ matrix.branch }}
+          submodules: recursive
+          fetch-depth: 0
+      - name: Add upstream
+        run: git remote add upstream https://github.com/solana-labs/solana.git
+      - name: Fetch upstream
+        run: git fetch upstream ${{ matrix.branch }}
+      - name: Set REBASE_BRANCH
+        run: echo "REBASE_BRANCH=ci/nightly/${{ matrix.branch }}/$(date +'%Y-%m-%d-%H-%M')" >> $GITHUB_ENV
```
Comment (esemeniuc) [inline .github/workflows/rebase.yaml:44]: use env keyword? https://docs.github.com/en/actions/learn-github-actions/variables#defining-environment-variables-for-a-single-workflow

File: .github/workflows/rebase.yaml [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+        timeout-minutes: 300
+        run: |
+          while true; do
+            response=$(curl -s -f -H "Authorization: Bearer ${{ secrets.BUILDKITE_TOKEN }}" "https://api.buildkite.com/v2/organizations/jito/pipelines/jito-solana/builds?commit=${{ env.REBASE_SHA }}")
+            if [ $? -ne 0 ]; then
+              echo "Curl request failed."
+              exit 1
+            fi
+
+            state=$(echo $response | jq -r '.[0].state')
```
Comment (esemeniuc) [inline .github/workflows/rebase.yaml:104]: might want to use `jq --exit-status` to exit if cant find var

**Review by esemeniuc — APPROVED** [review]

**Pull request merged**